universe/syncer.go

```diff
@@ -116,9 +122,31 @@ func (s *SimpleSyncer) executeSync(ctx context.Context, diffEngine DiffEngine,
 	// remote servers.
 	case globalInsertEnabled:
 		log.Infof("Fetching all roots for remote Universe server...")
-		targetRoots, err = diffEngine.RootNodes(ctx, false)
-		if err != nil {
-			return nil, err
+
+		offset := int32(0)
+
+		for {
+			log.Debugf("Fetching roots in range: %v to %v", offset,
+				offset+defaultPageLimit)
+			tempRoots, err := diffEngine.RootNodes(
```

Comment:
I think we can extract this into a `fetchAllRoots` function, which also sets us up to be able to add more unit tests in an isolated fashion later.